studio/http_provider.py

```diff
@@ -145,13 +154,7 @@ def checkpoint_experiment(self, experiment):
         self._raise_detailed_error(request)
         artifacts = request.json()['artifacts']
 
-        for tag, art in experiment.artifacts.iteritems():
-            if 'local' in art.keys():
-                art['key'] = artifacts[tag]['key']
```

Comment:
I could not find these keys added to the s3 storage implementation code. Are these keys still generated for all storage types?

Reply:
I am not sure I understand the question... This logic is specific to the db provider that talks to api server; and it simply uploads the file using the provided url. Does that somehow break the behavior that you expect?

Reply:
Thanks, answered your your comment in that this is for the DB provider not for the JSON queue messages.  Thanks